[2025-08-02T04:28:17.697+0000] {logging_mixin.py:188} INFO - Changing /opt/***/logs/dag_id=cdr_pipeline/run_id=manual__2025-08-02T04:05:53.026388+00:00/task_id=load_to_redshift permission to 509
[2025-08-02T04:28:17.838+0000] {logging_mixin.py:188} INFO - Changing /opt/***/logs/dag_id=cdr_pipeline/run_id=manual__2025-08-02T04:05:53.026388+00:00/task_id=load_to_redshift permission to 509
[2025-08-02T04:28:17.993+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: cdr_pipeline.load_to_redshift manual__2025-08-02T04:05:53.026388+00:00 [queued]>
[2025-08-02T04:28:18.046+0000] {taskinstance.py:1956} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: cdr_pipeline.load_to_redshift manual__2025-08-02T04:05:53.026388+00:00 [queued]>
[2025-08-02T04:28:18.051+0000] {taskinstance.py:2170} INFO - Starting attempt 6 of 6
[2025-08-02T04:28:18.124+0000] {taskinstance.py:2191} INFO - Executing <Task(S3ToRedshiftOperator): load_to_redshift> on 2025-08-02 04:05:53.026388+00:00
[2025-08-02T04:28:18.171+0000] {standard_task_runner.py:60} INFO - Started process 3783 to run task
[2025-08-02T04:28:18.192+0000] {standard_task_runner.py:87} INFO - Running: ['***', 'tasks', 'run', 'cdr_pipeline', 'load_to_redshift', 'manual__2025-08-02T04:05:53.026388+00:00', '--job-id', '150', '--raw', '--subdir', 'DAGS_FOLDER/dag.py', '--cfg-path', '/tmp/tmponxkx48_']
[2025-08-02T04:28:18.205+0000] {standard_task_runner.py:88} INFO - Job 150: Subtask load_to_redshift
[2025-08-02T04:28:18.367+0000] {logging_mixin.py:188} INFO - Changing /opt/***/logs/dag_id=cdr_pipeline/run_id=manual__2025-08-02T04:05:53.026388+00:00/task_id=load_to_redshift permission to 509
[2025-08-02T04:28:18.396+0000] {task_command.py:423} INFO - Running <TaskInstance: cdr_pipeline.load_to_redshift manual__2025-08-02T04:05:53.026388+00:00 [running]> on host f2527419f673
[2025-08-02T04:28:18.639+0000] {taskinstance.py:2480} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='cdr_pipeline' AIRFLOW_CTX_TASK_ID='load_to_redshift' AIRFLOW_CTX_EXECUTION_DATE='2025-08-02T04:05:53.026388+00:00' AIRFLOW_CTX_TRY_NUMBER='6' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-08-02T04:05:53.026388+00:00'
[2025-08-02T04:28:18.663+0000] {base.py:83} INFO - Using connection ID 'aws_default' for task execution.
[2025-08-02T04:28:18.680+0000] {base.py:83} INFO - Using connection ID 'aws_default' for task execution.
[2025-08-02T04:28:18.684+0000] {connection_wrapper.py:378} INFO - AWS Connection (conn_id='aws_default', conn_type='aws') credentials retrieved from login and password.
[2025-08-02T04:28:18.716+0000] {s3_to_redshift.py:192} INFO - Executing COPY command...
[2025-08-02T04:28:18.732+0000] {base.py:83} INFO - Using connection ID 'redshift_default' for task execution.
[2025-08-02T04:28:18.733+0000] {connection_wrapper.py:378} INFO - AWS Connection (conn_id='redshift_default', conn_type='aws') credentials retrieved from login and password.
[2025-08-02T04:28:18.734+0000] {warnings.py:109} WARNING - <string>:9: AirflowProviderDeprecationWarning: Host tf-redshift-cluster.ci896s3qzt6h.eu-west-1.redshift.amazonaws.com:5439/dev_db specified in the connection is not used. Please, set it on extra['endpoint_url'] instead

[2025-08-02T04:28:21.031+0000] {redshift_data.py:113} INFO - Polling statement fb30f1ed-4eda-4a0d-b7eb-b75640372396
[2025-08-02T04:28:21.220+0000] {redshift_data.py:129} INFO - Query STARTED
[2025-08-02T04:28:31.235+0000] {redshift_data.py:113} INFO - Polling statement fb30f1ed-4eda-4a0d-b7eb-b75640372396
[2025-08-02T04:28:31.454+0000] {taskinstance.py:2698} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/models/taskinstance.py", line 433, in _execute_task
    result = execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/amazon/aws/transfers/s3_to_redshift.py", line 194, in execute
    redshift_hook.execute_query(sql=sql, **self.redshift_data_api_kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/amazon/aws/hooks/redshift_data.py", line 107, in execute_query
    self.wait_for_results(statement_id, poll_interval=poll_interval)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/amazon/aws/hooks/redshift_data.py", line 124, in wait_for_results
    raise ValueError(
ValueError: Statement 'fb30f1ed-4eda-4a0d-b7eb-b75640372396' terminated with status FAILED. Response details: {'ClusterIdentifier': 'tf-redshift-cluster',
 'CreatedAt': datetime.datetime(2025, 8, 2, 4, 28, 22, 282000, tzinfo=tzlocal()),
 'Database': 'dev_db',
 'DbUser': 'masterdb',
 'Duration': -1,
 'Error': 'ERROR: Spectrum Scan Error\n'
          '  Detail: \n'
          '  -----------------------------------------------\n'
          '  error:  Spectrum Scan Error\n'
          '  code:      15007\n'
          '  context:   File '
          "'https://s3.eu-west-1.amazonaws.com/cdr-faker-data-dfad0cb7/cdr_2025-08-02.parquet/6dd739989123489bb43f109069ad0b1e.snappy.parquet' "
          'has an incompatible Parquet schema for column '
          "'s3://cdr-faker-data-dfad0cb7/cdr_2025-08-02.parquet/.roaming_status'. "
          'Co\n'
          '  query:     210276\n'
          '  location:  dory_util.cpp:1776\n'
          '  process:   worker_thread [pid=22947]\n'
          '  -----------------------------------------------\n',
 'HasResultSet': False,
 'Id': 'fb30f1ed-4eda-4a0d-b7eb-b75640372396',
 'QueryString': '\n'
                '                    COPY public.cdr_records \n'
                '                    FROM '
                "'s3://cdr-faker-data-dfad0cb7/cdr_2025-08-02.parquet/'\n"
                '                    credentials\n'
                "                    ''\n"
                '                    \n'
                '                    FORMAT AS PARQUET;\n'
                '        ',
 'RedshiftPid': 1073979673,
 'RedshiftQueryId': 210277,
 'ResponseMetadata': {'HTTPHeaders': {'content-length': '1182',
                                      'content-type': 'application/x-amz-json-1.1',
                                      'date': 'Sat, 02 Aug 2025 04:28:33 GMT',
                                      'x-amzn-requestid': 'b007ef8e-f47a-46f7-8804-7df394e2272d'},
                      'HTTPStatusCode': 200,
                      'RequestId': 'b007ef8e-f47a-46f7-8804-7df394e2272d',
                      'RetryAttempts': 0},
 'ResultRows': -1,
 'ResultSize': -1,
 'Status': 'FAILED',
 'UpdatedAt': datetime.datetime(2025, 8, 2, 4, 28, 23, 200000, tzinfo=tzlocal())}
[2025-08-02T04:28:31.473+0000] {taskinstance.py:1138} INFO - Marking task as FAILED. dag_id=cdr_pipeline, task_id=load_to_redshift, execution_date=20250802T040553, start_date=20250802T042818, end_date=20250802T042831
[2025-08-02T04:28:31.496+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 150 for task load_to_redshift (Statement 'fb30f1ed-4eda-4a0d-b7eb-b75640372396' terminated with status FAILED. Response details: {'ClusterIdentifier': 'tf-redshift-cluster',
 'CreatedAt': datetime.datetime(2025, 8, 2, 4, 28, 22, 282000, tzinfo=tzlocal()),
 'Database': 'dev_db',
 'DbUser': 'masterdb',
 'Duration': -1,
 'Error': 'ERROR: Spectrum Scan Error\n'
          '  Detail: \n'
          '  -----------------------------------------------\n'
          '  error:  Spectrum Scan Error\n'
          '  code:      15007\n'
          '  context:   File '
          "'https://s3.eu-west-1.amazonaws.com/cdr-faker-data-dfad0cb7/cdr_2025-08-02.parquet/6dd739989123489bb43f109069ad0b1e.snappy.parquet' "
          'has an incompatible Parquet schema for column '
          "'s3://cdr-faker-data-dfad0cb7/cdr_2025-08-02.parquet/.roaming_status'. "
          'Co\n'
          '  query:     210276\n'
          '  location:  dory_util.cpp:1776\n'
          '  process:   worker_thread [pid=22947]\n'
          '  -----------------------------------------------\n',
 'HasResultSet': False,
 'Id': 'fb30f1ed-4eda-4a0d-b7eb-b75640372396',
 'QueryString': '\n'
                '                    COPY public.cdr_records \n'
                '                    FROM '
                "'s3://cdr-faker-data-dfad0cb7/cdr_2025-08-02.parquet/'\n"
                '                    credentials\n'
                "                    ''\n"
                '                    \n'
                '                    FORMAT AS PARQUET;\n'
                '        ',
 'RedshiftPid': 1073979673,
 'RedshiftQueryId': 210277,
 'ResponseMetadata': {'HTTPHeaders': {'content-length': '1182',
                                      'content-type': 'application/x-amz-json-1.1',
                                      'date': 'Sat, 02 Aug 2025 04:28:33 GMT',
                                      'x-amzn-requestid': 'b007ef8e-f47a-46f7-8804-7df394e2272d'},
                      'HTTPStatusCode': 200,
                      'RequestId': 'b007ef8e-f47a-46f7-8804-7df394e2272d',
                      'RetryAttempts': 0},
 'ResultRows': -1,
 'ResultSize': -1,
 'Status': 'FAILED',
 'UpdatedAt': datetime.datetime(2025, 8, 2, 4, 28, 23, 200000, tzinfo=tzlocal())}; 3783)
[2025-08-02T04:28:31.554+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2025-08-02T04:28:31.580+0000] {taskinstance.py:3280} INFO - 0 downstream tasks scheduled from follow-on schedule check
